app/src/main/java/com/android/wildex/ui/social/UserSearchBar.kt

```diff
@@ -0,0 +1,179 @@
+          SearchBarDefaults.InputField(
+              modifier = Modifier.height(screenHeight / 45).testTag(SearchBarTestTags.INPUT_FIELD),
+              colors =
+                  SearchBarDefaults.colors()
+                      .inputFieldColors
+                      .copy(
+                          focusedTextColor = colorScheme.onBackground,
+                          unfocusedTextColor = colorScheme.onBackground,
+                          focusedContainerColor = colorScheme.surfaceVariant,
+                          unfocusedContainerColor = colorScheme.surfaceVariant,
+                          unfocusedLeadingIconColor = colorScheme.primary,
+                      ),
+              query = textFieldState.text.toString(),
+              onQueryChange = { textFieldState.edit { replace(0, length, it) } },
+              onSearch = { expanded = false },
+              expanded = expanded,
+              onExpandedChange = { expanded = it },
+              placeholder = { Text("Search users", style = typography.bodyMedium) },
+              leadingIcon = {
+                if (!expanded) {
+                  Icon(Icons.Default.Search, contentDescription = "Search")
+                } else {
+                  IconButton(
+                      modifier = Modifier.testTag(SearchBarTestTags.LEADING_ICON),
+                      onClick = { expanded = false }) {
+                        Icon(Icons.Default.ChevronLeft, contentDescription = "Go Back")
+                      }
+                }
+              },
+              trailingIcon = {
+                if (expanded) {
+                  IconButton(
+                      modifier = Modifier.testTag(SearchBarTestTags.TRAILING_ICON),
+                      onClick = { textFieldState.edit { delete(0, textFieldState.text.length) } }) {
+                        Icon(Icons.Default.Clear, contentDescription = "Clear text")
+                      }
+                }
+              })
+        },
+        expanded = expanded,
+        onExpandedChange = { expanded = it },
+    ) {
+      LazyColumn(modifier = Modifier.testTag(SearchBarTestTags.RESULT_LIST)) {
+        items(count = searchResults.size) { index ->
+          val user = searchResults[index]
+          ListItem(
+              headlineContent = {
+                Text(user.name + " " + user.surname, style = typography.bodyLarge)
+              },
+              supportingContent = {
+                Text(
+                    text = user.username,
+                    modifier =
+                        Modifier.testTag(SearchBarTestTags.testTagForResultUsername(user.userId)),
+                    style = typography.bodyMedium)
+              },
+              leadingContent = {
+                ClickableProfilePicture(
+                    modifier = Modifier.size(45.dp),
+                    profilePictureURL = user.profilePictureURL,
+                    profileUserType = user.userType,
+                )
```

Comment:
Currently, one is only taken to the profile page of the user when clicking on the bigger `ListItem` composable: clicking directly on the profile picture does not work. We just need to remember to give the same parameters to the `ClickableProfilePicture` composable to fully use it. 

```suggestion
                ClickableProfilePicture(
                    modifier = Modifier.size(45.dp),
                    profileId = user.userId,
                    profilePictureURL = user.profilePictureURL,
                    profileUserType = user.userType,
                    onProfile = onResultClick,
                )
```

Reply:
Good catch, I did it on purpose because I thought It would be redundant to have it in both and I thought setting it for the list item would also work when clicking the profile picture, as it was a child of the list item. But I forgot to test it... Thanks for seeing that!